calicoctl/commands/file/iter.go

```diff
@@ -0,0 +1,73 @@
+package file
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// Iter extracts the filename from the parsed args and
+// -  invokes the callback for each manifest files in the directory if the filename in the parsed arguments is a
+//    directory (updating the arguments to include the specific file)
+// -  otherwise just invoke the callback with the unmodified arguments.
+func Iter(parsedArgs map[string]interface{}, cb func(map[string]interface{}) error) error {
+	if parsedArgs["--filename"] == nil {
+		// Handle no filename being specified by invoking standard processing.
+		return cb(parsedArgs)
+	}
+
+	// File name is specified.
+	f, ok := parsedArgs["--filename"].(string)
+	if !ok {
+		// Handle invalid filename by invoking standard processing.
+		return cb(parsedArgs)
+	}
+
+	if parsedArgs["--filename"] == "-" {
+		// Handle stdin filename by invoking standard processing.
+		return cb(parsedArgs)
+	}
+
+	if !isDir(f) {
+		// Handle non-directory by invoking standard processing.
+		return cb(parsedArgs)
+	}
+
+	// Handle directory by walking the directory contents and performing the action on each manifest file.
+	return filepath.Walk(f,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if info.IsDir() {
+				return nil
+			}
```

Comment:
Since we aren't returning a SkipDir here, this means we will process this directory as well?

Reply:
Oooh that was not intentional.  I will fix.